File: Python/programmers/etc/level1/햄버거_만들기.py
def solution(ingredient):
    answer = 0
    stack = []
    for i in ingredient:
        stack.append(i)
        if stack[-4:] == [1,2,3,1]:
            for _ in range(4):
                stack.pop()
            answer += 1
    return answer


ingredient = [2, 1, 1, 2, 3, 1, 2, 3, 1]
print(solution(ingredient)) # 2

ingredient = [1, 3, 2, 1, 2, 1, 3, 1, 2]
print(solution(ingredient)) # 0

File: Java/programmers/etc/level1/햄버거만들기.java
package CodeTest.Java.programmers.etc.level1;

public class 햄버거만들기 {
    public int solution(int[] ingredient) {
        int answer = 0;
        int[] stk = new int[ingredient.length];

        int j = 0;
        for (int i : ingredient) {
            stk[j++] = i;
            if (j >= 4 && stk[j-1]==1 && stk[j-2]==3 && stk[j-3]==2 && stk[j-4]==1) {
                j -= 4;
                answer++;
            }
        }
        
        return answer;
    }
}
